pkg/util/vsphere/vm/cdrom.go

```diff
@@ -0,0 +1,388 @@
+// Copyright (c) 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package vm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/vmware/govmomi/object"
+	vimtypes "github.com/vmware/govmomi/vim25/types"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	imgregv1a1 "github.com/vmware-tanzu/image-registry-operator-api/api/v1alpha1"
+
+	vmopv1 "github.com/vmware-tanzu/vm-operator/api/v1alpha3"
+	pkgctx "github.com/vmware-tanzu/vm-operator/pkg/context"
+	"github.com/vmware-tanzu/vm-operator/pkg/util"
+)
+
+const (
+	vmiKind  = "VirtualMachineImage"
+	cvmiKind = "Cluster" + vmiKind
+
+	serverIDSuffix = "?serverId="
+)
+
+// UpdateCdromDeviceChanges reconciles the desired CD-ROM devices specified in
+// VM.Spec.Cdrom with the current CD-ROM devices in the VM. It returns a list of
+// device changes required to update the CD-ROM devices.
+func UpdateCdromDeviceChanges(
+	vmCtx pkgctx.VirtualMachineContext,
+	client ctrlclient.Client,
+	curDevices object.VirtualDeviceList) ([]vimtypes.BaseVirtualDeviceConfigSpec, error) {
+	var (
```

Comment:
Please leave a blank line between the function signature and the `var` statement.